packages/ns-api/files/ns.reverseproxy

```diff
@@ -88,245 +100,321 @@ if cmd == 'list':
-                        server = servers[entry['uci_server']]
-                        item = {
-                            'id': entry['uci_server'],
-                            'type': 'domain',
-                            'description': server['uci_description'],
-                            'location': entry['location'],
-                            'domain': server['server_name'],
-                            'destination': entry['proxy_pass']
-                        }
-                        for cert_name, cert_path in __certificate_list().items():
-                            if cert_path + '.crt' == server['ssl_certificate']:
-                                item['certificate'] = cert_name
-                        if 'allow' in server:
-                            item['allow'] = server['allow']
-                        response.append(item)
-
-                print(json.dumps({'data': response}))
-
-            elif action == 'edit-path':
-                # load available paths
-                paths = {entry_id: entry for (entry_id, entry)
+        if action == 'list-certificates':
+            default_cert = e_uci.get('nginx', '_lan', 'ssl_certificate', default='')
+            default_cert = default_cert.removesuffix('.crt')
+            print(json.dumps({
+                'values': list(__certificate_list().keys()),
+                'default': default_cert.rsplit('/')[-1]
+            }))
+
+        if action == 'add-certificate':
+            data = json.loads(sys.stdin.read())
+            if 'name' not in data:
+                raise ValidationError('name', 'required')
+            if 'name' in __certificate_list():
+                raise ValidationError('name', 'unique', data['name'])
+            if utils.sanitize(data['name']) != data['name']:
+                raise ValidationError('name', 'invalid', data['name'])
+            if 'certificate_path' not in data:
+                raise ValidationError('certificate_path', 'required')
+            if os.path.isfile(data['certificate_path']) is False:
+                raise ValidationError('certificate_path', 'invalid', data['certificate_path'])
+            if 'key_path' not in data:
+                raise ValidationError('key_path', 'required')
+            if os.path.isfile(data['key_path']) is False:
+                raise ValidationError('key_path', 'invalid', data['key_path'])
+            if 'chain_path' in data and os.path.isfile(data['chain_path']) is False:
+                raise ValidationError('chain_path', 'invalid', data['chain_path'])
+
+            # store certificate
+            with open(data['certificate_path'], 'r') as cert_file:
+                with open(f'/etc/acme/{data["name"]}.crt', 'a') as cert:
+                    cert.write(cert_file.read())
+
+                if 'chain_path' in data:
+                    with open(data['chain_path'], 'r') as chain_file:
+                        cert.write(chain_file.read())
```

Comment:
I prefer to write the certificate on new file, making sure it does not contain anything else.
```suggestion
            with open(data['certificate_path'], 'r') as cert_file:
                with open(f'/etc/acme/{data["name"]}.crt', 'w') as cert:
                    cert.write(cert_file.read())

            if 'chain_path' in data:
                with open(f'/etc/acme/{data["name"]}.crt', 'a') as cert:
                    with open(data['chain_path'], 'r') as chain_file:
                        cert.write(chain_file.read())
```